### ISSUE TITLE
Issue when build is called more than once

### ISSUE THREAD

**Issue by BenevidesLecontes** [BendixMa/sliding-sheet#3]
Hi, thank your for this amazing lib, I'm having issue when using routes and I have a component with sliding-sheet as initial route or home in the scaffold widget. As this issue [https://github.com/flutter/flutter/issues/23063](url), this is working as expected. I changed some lines in your lib and fixed the issue, If you wan't I can make PR for this. I changed these lines.

```
_childHeight = child?.size?.height ?? 0;
      _headerHeight = header?.size?.height ?? 0;
      _footerHeight = footer?.size?.height ?? 0;
      _availableHeight = parent?.size?.height ?? 0;
```

to

```
      _childHeight = child?.hasSize ? child.size.height : 0;
      _headerHeight = header?.hasSize ? header?.size.height : 0;
      _footerHeight = header?.hasSize ? footer.size?.height : 0;
      _availableHeight = parent?.hasSize ? parent?.size.height : 0;
```

The issue
```
════════ Exception caught by scheduler library ═════════════════════════════════════════════════════
The following assertion was thrown during a scheduler callback:
RenderBox was not laid out: RenderConstrainedBox#11af2 NEEDS-LAYOUT NEEDS-PAINT
'package:flutter/src/rendering/box.dart':
Failed assertion: line 1687 pos 12: 'hasSize'


Either the assertion indicates an error in the framework itself, or we should provide substantially more information in this error message to help you determine and fix the underlying cause.
In either case, please report this assertion by filing a bug on GitHub:
  https://github.com/flutter/flutter/issues/new?template=BUG.md

When the exception was thrown, this was the stack: 
#2      RenderBox.size (package:flutter/src/rendering/box.dart:1687:12)
#3      _SlidingSheetState._measure.<anonymous closure> (package:sliding_sheet/src/sheet.dart:348:29)
#4      postFrame.<anonymous closure> (package:sliding_sheet/src/util.dart:5:63)
#5      SchedulerBinding._invokeFrameCallback (package:flutter/src/scheduler/binding.dart:1102:15)
#6      SchedulerBinding.handleDrawFrame (package:flutter/src/scheduler/binding.dart:1049:9)
...
════════════════════════════════════════════════════════════════════════════════════════════════════
```

**Comment by BendixMa** [comment]
Hey, thanks for the report! I wasn't able to reproduce the issue however you're solution is probably the safer way to ensure the widget has been laid out. I released a new version `(0.2.5)` with those changes. Could you check if that solves the issue you're having?

**Comment by BenevidesLecontes** [comment]
@BendixMa thanks! It's working fine the new version.
